plugins/woocommerce/src/Blocks/BlockTypes/Reviews/ProductReviewContent.php

```diff
@@ -0,0 +1,93 @@
+<?php declare( strict_types = 1 );
+namespace Automattic\WooCommerce\Blocks\BlockTypes\Reviews;
+
+use Automattic\WooCommerce\Blocks\BlockTypes\AbstractBlock;
+
+/**
+ * ProductReviewContent class.
+ */
+class ProductReviewContent extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-review-content';
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 * @return string Rendered block content.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		if ( ! isset( $block->context['commentId'] ) ) {
+			return '';
+		}
+
+		$comment            = get_comment( $block->context['commentId'] );
+		$commenter          = wp_get_current_commenter();
+		$show_pending_links = isset( $commenter['comment_author'] ) && $commenter['comment_author'];
+		if ( empty( $comment ) ) {
+			return '';
+		}
+
+		$args         = array();
+		$comment_text = get_comment_text( $comment, $args );
+		if ( ! $comment_text ) {
+			return '';
+		}
+
+		/**
+		 * This filter is documented in wp-includes/comment-template.php
+		 *
+		 * @since 1.2.0
+		 */
+		$comment_text = apply_filters( 'comment_text', $comment_text, $comment, $args );
+
+		$moderation_note = '';
+		if ( '0' === $comment->comment_approved ) {
+			$commenter = wp_get_current_commenter();
+
+			if ( $commenter['comment_author_email'] ) {
+				$moderation_note = __( 'Your review is awaiting moderation.', 'woocommerce' );
+			} else {
+				$moderation_note = __( 'Your review is awaiting moderation. This is a preview; your review will be visible after it has been approved.', 'woocommerce' );
+			}
+			$moderation_note = '<p><em class="review-awaiting-moderation">' . $moderation_note . '</em></p>';
```

Comment:
As think we should use `esc_html` here as the `$moderation_note` is just a string.

Reply:
I've wrapped `$moderation_note` in `esc_html` in https://github.com/woocommerce/woocommerce/pull/57533/commits/3d2ec2574be1c01c40d72bf08b6108ae34006b5e. Let me know if I've misunderstood!